src/main/kotlin/no/fdk/dataset_catalog/extensions/DatasetExtensions.kt

```diff
@@ -17,7 +21,7 @@ fun Dataset.updateSubjects(): Dataset =
         )
     })
 
-fun Dataset.update(newValues: Dataset): Dataset =
+fun Dataset.update(newValues: DatasetDTO): Dataset =
```

Comment:
Siden du uansett hadde behov for `.toDBO()` så trenger du egentlig ikke gjøre endringer her, bare map til dbo før `.update()`. Men same-same, gjør det du føler er mest leselig.